src/control/cmd/daos/container.go

```diff
@@ -361,6 +363,86 @@ func (cmd *containerCreateCmd) Execute(_ []string) (err error) {
 	return nil
 }
 
+type containerSerializeCmd struct {
+	daosCmd
+
+	Output string `long:"output-path" short:"o" description:"path to output serialized HDF5 files" required:"0"`
+	Args   struct {
+		Source string `positional-arg-name:"<Source container>"`
+	} `positional-args:"yes"`
+}
+
+func (cmd *containerSerializeCmd) Execute(_ []string) error {
+	ap, deallocCmdArgs, err := allocCmdArgs(cmd.log)
+	if err != nil {
+		return err
+	}
+	if cmd.Args.Source == "" {
+		return errors.New("Source container is required")
+	}
+	defer deallocCmdArgs()
+
+	ap.src = C.CString(cmd.Args.Source)
+	defer freeString(ap.src)
+	if cmd.Output != "" {
+		ap.output_path = C.CString(cmd.Output)
+		defer freeString(ap.output_path)
+	}
+
+	ap.c_op = C.CONT_SERIALIZE
+	rc := C.cont_serialize_hdlr(ap)
+	if err := daosError(rc); err != nil {
+		return errors.Wrapf(err,
+			"failed to serialize %s", cmd.Args.Source)
+	}
+
+	return nil
+}
+
+type containerDeserializeCmd struct {
+	daosCmd
+
+	Label string `long:"label" short:"l" description:"Create container with specified container label" required:"0"`
+	Args  struct {
+		Pool string `positional-arg-name:"<Pool UUID or Label>"`
+		File string `positional-arg-name:"<File to deserialize>"`
+	} `positional-args:"yes"`
+}
+
+func (cmd *containerDeserializeCmd) Execute(_ []string) error {
+	ap, deallocCmdArgs, err := allocCmdArgs(cmd.log)
+	if err != nil {
+		return err
+	}
+	if cmd.Args.Pool == "" {
+		return errors.New("Pool is required")
+	}
+	if cmd.Args.File == "" {
+		return errors.New("File to deserialize is required")
+	}
+	defer deallocCmdArgs()
+
+	cPool := C.CString(cmd.Args.Pool)
+	defer freeString(cPool)
+	C.strncpy(&ap.pool_str[0], cPool, C.DAOS_PROP_LABEL_MAX_LEN)
+	ap.path = C.CString(cmd.Args.File)
+	defer freeString(ap.path)
+	if cmd.Label != "" {
+		cLabel := C.CString(cmd.Label)
+		defer freeString(cLabel)
+		C.strncpy(&ap.cont_str[0], cLabel, C.DAOS_PROP_LABEL_MAX_LEN)
```

Comment:
as above, where does cont_str get freed, maybe we need to add free checks for these in deallocCmdArgs?